### ISSUE TITLE
When angular2 and typescript is upgraded, npm start is not building

### ISSUE THREAD

**Issue by munavargit** [datatypevoid/vulgar#40]
I have changed the the 2.0.0 rc1 version of angular2 modules to latest and typescript from ~1.8.9 to ^2.0.0. NPM Start is not building the project.

Package.json changes:

in Dependencies,

...
"@angular/common": "~2.4.0",
    "@angular/compiler": "~2.4.0",
    "@angular/core": "~2.4.0",
    "@angular/forms": "~2.4.0",
    "@angular/http": "~2.4.0",
    "@angular/platform-browser": "~2.4.0",
    "@angular/platform-browser-dynamic": "~2.4.0",
    "@angular/router": "~3.4.0",
	"@angular/platform-server": "2.0.0-rc.1",   
    "@angular/router-deprecated": "2.0.0-rc.1",
......

in devDependencies,

"typescript": "^2.0.0",

above are the changes. (without these changes.. all works just fine, however, I would like to upgrade to latst angular2 version and typescript version).

The below are the error we get,

62% 191/218 build modules[default] File 'D:/Source-Code/sample-mean2/sample1/no
de_modules/core-js/es6/index.js' has unsupported extension. The only supported e
xtensions are '.ts', '.tsx', '.d.ts'.
[default] File 'D:/Source-Code/sample-mean2/sample1/node_modules/core-js/es7/ref
lect.js' has unsupported extension. The only supported extensions are '.ts', '.t
sx', '.d.ts'.
[default] File 'D:/Source-Code/sample-mean2/sample1/node_modules/source-map/sour
ce-map.js' has unsupported extension. The only supported extensions are '.ts', '
.tsx', '.d.ts'.
[default] File 'D:/Source-Code/sample-mean2/sample1/node_modules/uglify-js/tools
/node.js' has unsupported extension. The only supported extensions are '.ts', '.
tsx', '.d.ts'.
[default] D:/Source-Code/sample-mean2/sample1/node_modules/@angular/router-depre
cated/src/directives/router_outlet.d.ts:2:10
    Module '"D:/Source-Code/sample-mean2/sample1/node_modules/@angular/core/inde
x"' has no exported member 'DynamicComponentLoader'.
[default] D:/Source-Code/sample-mean2/sample1/node_modules/@angular2-material/bu
tton/button.d.ts:22:44
    Generic type 'Type<T>' requires 1 type argument(s).
[default] D:/Source-Code/sample-mean2/sample1/node_modules/@angular2-material/co
re/portal/portal-directives.d.ts:1:37
    Module '"D:/Source-Code/sample-mean2/sample1/node_modules/@angular/core/inde
x"' has no exported member 'ComponentResolver'.
[default] D:/Source-Code/sample-mean2/sample1/node_modules/@angular2-material/co
re/portal/portal.d.ts:25:16
    Generic type 'Type<T>' requires 1 type argument(s).
[default] D:/Source-Code/sample-mean2/sample1/node_modules/@angular2-material/co
re/portal/portal.d.ts:32:28
    Generic type 'Type<T>' requires 1 type argument(s).
[default] D:/Source-Code/sample-mean2/sample1/src/app/app.component.ts:41:3
    Argument of type '{ selector: string; providers: undefined[]; directives: (t
ypeof NgFor | typeof RouterActive | typ...' is not assignable to parameter of ty
pe 'Component'.
  Object literal may only specify known properties, and 'directives' does not ex
ist in type 'Component'.
[default] D:/Source-Code/sample-mean2/sample1/src/app/home/home.component.ts:26:
3
    Argument of type '{ selector: string; providers: typeof Title[]; directives:
 (typeof NgFor | typeof XLarge | typeof...' is not assignable to parameter of ty
pe 'Component'.
  Object literal may only specify known properties, and 'directives' does not ex
ist in type 'Component'.
[default] D:/Source-Code/sample-mean2/sample1/src/app/recipes/rating.component.t
s:18:3
    Argument of type '{ selector: string; template: any; directives: undefined[]
; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not ex
ist in type 'Component'.
[default] D:/Source-Code/sample-mean2/sample1/src/app/recipes/recipe-details.com
ponent.ts:29:3
    Argument of type '{ selector: string; template: any; directives: typeof Rati
ng[]; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not ex
ist in type 'Component'.
....
....
....

Please advise to update MEAN2 with latest Angular2 and Typescript version.

**Comment by NickBolles** [comment]
This should be covered with my PR #41

**Comment by munavargit** [comment]
That's fantastic, please update when done

Thanks.

**Comment by NickBolles** [comment]
It should be ready to go. But it is based off the development branch, so its quite a bit different

**Comment by munavargit** [comment]
@NickBolles ,

Yes, I saw, even Node App is changed changed to TypeScript (I just realized it's was done in November 2016). 

I have downloaded the clone from "https://github.com/NickBolles/vulgar/tree/nb/materialize"

It is possible to use just Client (Augular2 Application) and continue to use old node app (master), because, I have completed all the coding with my node app?

**Comment by NickBolles** [comment]
Hey, I apologize, I didn't get a notification for some reason. Yea it should be possible. It might be easiest to convert your app.js file into a typescript file though. You could do something as simple as 
`import './app.js'`
in an `app.ts` file

This is just a guess, I am not sure how the config for your app works and how it has changed. I just found vulgar last week and am still understanding how it all fits together.

The biggest thing is probably going to be how the database and config is setup. In this version the env.conf.ts and mongoose.conf.ts as well as the passport.conf.ts do most of the connecting and configuring. This might be different from how your server is doing it, but I'm sure you can just use what you have in place already for now.

**Comment by munavargit** [comment]
Hi,

Thanks for your support.

However, I download another latest Angular2 boiler plate, and integrated my
existing Node app code with it to proceed.

Thanks.

**Comment by NickBolles** [comment]
May I ask which boiler plate your using?

Get Outlook for Android<https://aka.ms/ghei36>


________________________________

**Comment by munavargit** [comment]
Yes, sure.

I'm using https://github.com/AngularClass/angular2-webpack-starter. It uses
typescript 2.1.6 and Angular 2.4.6.
